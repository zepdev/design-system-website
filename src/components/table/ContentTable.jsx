import React, { useState } from 'react'
import PropTypes from 'prop-types'
import withStyles from 'react-jss'
import clsx from 'clsx'
import Table from './Table'
import TableHeader from './TableHeader'
import TableCell from './TableCell'
import TableBody from './TableBody'
import TableRow from './TableRow'
import Tabs from '../tabs/Tabs'
import Tab from '../tabs/Tab'

const styles = theme => ({
  root: {
    overflowX: 'auto',
  },
  table: {
    marginBottom: `${ theme.spacing.component.xxl.rem }rem`,
  },
  lowercase: {
    textTransform: 'lowercase',
  },
  capitalize: {
    textTransform: 'capitalize',
  },
  tabs: {
    marginBottom: `${ theme.spacing.component.l.rem }rem`,
  },
  inverse: {
    background: theme.colors.gray.black.hex,
    color: 'rgba(255, 255, 255, 0.84)',
  },
  inverseLight: {
    background: theme.colors.gray.grayMidDark.hex,
    color: 'rgba(255, 255, 255, 0.84)',
  },
})

const ContentTable = ({ header, content, title, classes }) => {
  const [value, setValue] = useState(0)

  function handleChange(newValue) {
    setValue(newValue)
  }

  return (
    <>
      <Tabs value={value} onClick={handleChange} className={classes.tabs}>
        <Tab label="Default" />
        <Tab label="Inverse" />
      </Tabs>
      <div className={classes.root}>
        <Table className={classes.table} title={title}>
          <TableHeader>
            {header.map(elem => (
              <TableCell
                className={clsx({
                  [classes.lowercase]: elem === 'rem' || elem === 'px',
                  [classes.capitalize]: elem !== 'rem' && elem !== 'px',
                  [classes.inverse]: value === 1,
                })}
                key={`header_${ elem }`}
              >
                {elem}
              </TableCell>
            ))}
          </TableHeader>
          {content && (
            <TableBody>
              {Object.keys(content).map((elem, idx) => (
                <TableRow
                  key={`row_${ elem }`}
                  color={idx % 2 === 0 ? 'white' : 'gray'}
                  className={
                    value === 1
                      ? idx % 2 === 0
                        ? classes.inverseLight
                        : classes.inverse
                      : null
                  }
                >
                  {header.map((item, idx) =>
                    idx === 0 ? (
                      <TableCell
                        key={`typography${ idx }`}
                        scope="row"
                        component="th"
                      >
                        {item === 'actual size' ? (
                          'lorem ipsum'
                        ) : item === 'name' ? (
                          elem
                        ) : item === 'specifications' ? (
                          <span>
                            {content[elem][item].map(p => (
                              <p
                                key={`specification_${ p }`}
                                className={
                                  value === 1
                                    ? 'zep-typo--normal-body2-inverse'
                                    : 'zep-typo--normal-body2'
                                }
                              >
                                {p}
                              </p>
                            ))}
                          </span>
                        ) : content[elem][item] ? (
                          content[elem][item]
                        ) : (
                          '-'
                        )}
                      </TableCell>
                    ) : (
                      <TableCell
                        className={clsx({
                          [elem]: item === 'actual size' && value === 0,
                          [`${ elem }-inverse`]:
                            item === 'actual size' && value === 1,
                        })}
                        key={`typography${ idx }`}
                      >
                        {item === 'actual size' ? (
                          'lorem ipsum'
                        ) : item === 'name' ? (
                          elem
                        ) : item === 'specifications' ? (
                          <span>
                            {content[elem][item].map(p => (
                              <p
                                key={`specification_${ p }`}
                                className={
                                  value === 1
                                    ? 'zep-typo--normal-body2-inverse'
                                    : 'zep-typo--normal-body2'
                                }
                              >
                                {p}
                              </p>
                            ))}
                          </span>
                        ) : content[elem][item] ? (
                          content[elem][item]
                        ) : (
                          '-'
                        )}
                      </TableCell>
                    )
                  )}
                </TableRow>
              ))}
            </TableBody>
          )}
        </Table>
      </div>
    </>
  )
}

ContentTable.propTypes = {
  classes: PropTypes.object.isRequired,
  header: PropTypes.array.isRequired,
  content: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired,
}

export default withStyles(styles)(ContentTable)
