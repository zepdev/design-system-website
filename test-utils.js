import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@material-ui/core/styles'
import theme from './src/theme'

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
