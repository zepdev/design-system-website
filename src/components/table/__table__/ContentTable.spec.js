import React from 'react'
import { render } from '../../../../test-utils'
import ContentTable from '../ContentTable'

describe('ContentTable', () => {
  it('renders correctly', () => {
    const mockProps = {
      header: ['mockHeader1'],
      content: { item1: 'mockItem' },
      title: 'mockTitle',
    }
    const component = render(<ContentTable {...mockProps} />)
    expect(component).toMatchSnapshot()
  })
})
