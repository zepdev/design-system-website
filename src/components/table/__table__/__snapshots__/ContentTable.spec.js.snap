// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentTable renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="root-0-2-8 tabs-0-2-5 zep-tabs"
        data-testid="tabs"
        role="tablist"
      >
        <div
          class="zep-tabs__item"
        >
          <button
            aria-selected="true"
            class="zep-tab zep-tab--selected"
            id="tab-0"
            role="tab"
            tabindex="0"
          >
            Positive
          </button>
        </div>
        <div
          class="zep-tabs__item"
        >
          <button
            aria-selected="false"
            class="zep-tab"
            id="tab-1"
            role="tab"
            tabindex="0"
          >
            Negative
          </button>
        </div>
      </div>
      <div
        class="root-0-2-1"
      >
        <table
          class="zep-table table-0-2-2"
        >
          <caption
            class="zep-visually-hidden"
          >
            mockTitle
          </caption>
          <thead>
            <tr
              class="zep-table__row zep-table__row--header"
            >
              <th
                class="zep-table__cell zep-table__cell--header capitalize-0-2-4"
                scope="col"
              >
                mockHeader1
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              class="zep-table__row zep-table__row--white"
            >
              <th
                class="zep-table__cell"
                scope="row"
              >
                -
              </th>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="root-0-2-8 tabs-0-2-5 zep-tabs"
      data-testid="tabs"
      role="tablist"
    >
      <div
        class="zep-tabs__item"
      >
        <button
          aria-selected="true"
          class="zep-tab zep-tab--selected"
          id="tab-0"
          role="tab"
          tabindex="0"
        >
          Positive
        </button>
      </div>
      <div
        class="zep-tabs__item"
      >
        <button
          aria-selected="false"
          class="zep-tab"
          id="tab-1"
          role="tab"
          tabindex="0"
        >
          Negative
        </button>
      </div>
    </div>
    <div
      class="root-0-2-1"
    >
      <table
        class="zep-table table-0-2-2"
      >
        <caption
          class="zep-visually-hidden"
        >
          mockTitle
        </caption>
        <thead>
          <tr
            class="zep-table__row zep-table__row--header"
          >
            <th
              class="zep-table__cell zep-table__cell--header capitalize-0-2-4"
              scope="col"
            >
              mockHeader1
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="zep-table__row zep-table__row--white"
          >
            <th
              class="zep-table__cell"
              scope="row"
            >
              -
            </th>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
